api/src/main/java/com/xwiki/pdfviewer/macro/PDFViewerMacroParameters.java

```diff
@@ -0,0 +1,157 @@
+     * name of the attachment when is used along with the document parameter.
+     */
+    private String file;
+
+    /**
+     * The viewer width, in pixels. Uses a percentage value, for example: 25%, 50%, 100%.
+     */
+    private String width = "100%";
+
+    /**
+     * The viewer height, in pixels.
+     */
+    private int height = 1000;
+
+    /**
+     * String reference of the document that contains the file. It is ignored if file is an URL or the resulted
+     * attachment reference does not actually exists.
+     */
+    private String document;
+
+    /**
+     * If this value is true (or 1 or yes) and the user has no access to the  document to which the PDF file is
+     * attached, the PDF file could still be viewed on behalf of the author of the document containing the macro (if
+     * that author has access to the containing document). TODO: change the type to boolean and add a migration for the
+     * old values.
+     */
+    private String asAuthor = "0";
+
+    /**
+     * @return the reference to the PDF file, an absolute URL or simply the file name
+     */
+    public String getFile()
+    {
+        return this.file;
+    }
+
+    /**
+     * Set the value of the PDF file by using a full attachment reference, an absolute URL or simply the file name.
+     * 
+     * @param file the reference to the PDF file, an absolute URL or simply the file name
+     */
+    @PropertyDisplayType(PDFResourceReference.class)
+    @PropertyMandatory
+    public void setFile(String file)
+    {
+        this.file = file;
+    }
+
+    /**
+     * @return the display width of the PDF file
+     */
+    public String getWidth()
+    {
+        return width;
+    }
+
+    /**
+     * Set the display width of the PDF file.
+     * 
+     * @param width display width of the PDF file
+     */
+    public void setWidth(String width)
+    {
+        this.width = width;
+    }
+
+    /**
+     * @return the display height of the PDF file
+     */
+    public int getHeight()
+    {
+        return height;
+    }
+
+    /**
+     * Set the display height of the PDF file.
+     * 
+     * @param height display height of the PDF file
+     */
+    public void setHeight(int height)
+    {
+        this.height = height;
+    }
+
+    /**
+     * @return the reference of the document that contains the PDF file to be viewed
+     */
+    public String getDocument()
+    {
+        return document;
+    }
+
+    /**
+     * Set the reference of the document that contains the PDF file to be viewed.
+     * 
+     * @param document reference of the document that contains the PDF file
+     */
+    @PropertyAdvanced
+    public void setDocument(String document)
+    {
+        this.document = document;
+    }
+
+    /**
+     * @return true (or 1 or yes) if the view right of the PDF file document should be delegated to the users that
+     *         requests to see it through the macro, false (or 0 or no) otherwise
```

Comment:
```suggestion
     * @return whether to access the PDF file on behalf of the last author of the page calling the PDF Viewer macro
               (using their access rights) rather than on behalf of the current user (using the access rights of the
               current user)
```